src/bootstrap/src/lib.rs

```diff
@@ -707,8 +708,12 @@ impl Build {
 
     /// Gets the space-separated set of activated features for the standard
     /// library.
-    fn std_features(&self, target: TargetSelection) -> String {
-        let mut features = " panic-unwind".to_string();
+    fn std_features(&self, target: TargetSelection, stage: u32) -> String {
+        let mut features = match self.config.rust_panic_strategy_std {
+            PanicStrategy::Abort if stage > 0 => " panic_immediate_abort",
```

Comment:
I do not think that we should be doing this, aborting panics are not the same as `panic_immediate_abort`; aborting panics are effectively just panics that don't unwind the stack. They still do all the formatting of the panic message and print a backtrace. Immediate-abort panics don't do any formatting and just call `core::intrinsics::abort.

I have thought about ways to make immediate-abort a proper panic strategy, and one could do that albiet poorly by changing the lowering of calls to the panic lang items when the feature is enabled. That doesn't fix all the `cfg`s which are most of the maintenance burden.